Examples/SDWebImage Demo/DetailViewController.m

```diff
@@ -37,6 +38,39 @@ - (void)viewDidLoad {
                                                                             style:UIBarButtonItemStylePlain
                                                                            target:self
                                                                            action:@selector(toggleAnimation:)];
+    // Add a secret title click action to apply tint color
+    UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
+    [button addTarget:self
+               action:@selector(toggleTint:)
+     forControlEvents:UIControlEventTouchUpInside];
+    [button setTitle:@"Tint" forState:UIControlStateNormal];
+    self.navigationItem.titleView = button;
+}
+
+- (void)toggleTint:(UIResponder *)sender {
+    // tint for non-opaque animation
+    if (!self.imageView.isAnimating) {
+        return;
+    }
+    SDAnimatedImage *animatedImage = (SDAnimatedImage *)self.imageView.image;
+    if (animatedImage.sd_imageFormat == SDImageFormatGIF) {
+        // GIF is opaque
+        return;
+    }
+    BOOL containsAlpha = [SDImageCoderHelper CGImageContainsAlpha:animatedImage.CGImage];
+    if (!containsAlpha) {
+        return;
+    }
+    if (self.tintApplied) {
+        self.imageView.animationTransformer = nil;
+    } else {
+        self.imageView.animationTransformer = [SDImageTintTransformer transformerWithColor:UIColor.blackColor];
```

Comment:
@Kudo Usage code on `SDAnimatedImageView` with custom post-transform

Reply:
sorry was busy on releasing new stuff from the past week and late for the response.

i tried to apply this pr's patch locally and the new `animationTransformer` works well. i like this api and super excited to use it in formal.

just one more question: the `animationTransformer` only works for animated images now. could we have similar post-transform for static images?

Reply:
`could we have similar post-transform for static images?`

Static image itself supports transformer. You can pass context options `SDWebImageContextImageTransformer` to use that. It's not **per-imageView level` control, it's `per-image level` control.

Reply:
See: https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#image-transformer-50

Reply:
cool thanks for sharing the info. it's just a sense of inconsistent usage between static image and animated image. but no problem we can manage the difference internally. will try the new release in the next expo sdk. thanks for having all the great help.

Reply:
> inconsistent

The actual reason why this API looks inconsistent is because, we don't `Cache the animated image frame into global SDImageCache` at all. Which will result OOM, and really a bad design

So, the `animationTransformer` is applied to `image view level`, all the cache or status only effect one image view.
`transformer` is applied to `image object level`, by default only the `UIImage` (not `_UIAnimatedImage` or `SDAnimatedImage`)